x-pack/plugins/security_solution/server/lib/detection_engine/signals/bulk_create_ml_signals.ts

```diff
@@ -15,6 +14,7 @@ import { RuleTypeParams, RefreshTypes } from '../types';
 import { singleBulkCreate, SingleBulkCreateResponse } from './single_bulk_create';
 import { AnomalyResults, Anomaly } from '../../machine_learning';
 import { BuildRuleMessage } from './rule_messages';
+import { SearchResponse } from '../../types';
```

Comment:
I think I was playing around with the types as well and forgot to switch this one back...got curious about the duplicates

Reply:
Ah I got an error when changing it back and tracked it down. The search response for KQL query rules is being typed based on the custom `SearchResponse` type because it is accessing `hits.total` which looks to be typed (incorrectly) as `number` in the official `SearchResponse` type. To share the "filter against lists" code with the KQL rules then the ML bulk create logic has to accept our custom `SearchResponse`.

Need to investigate further why the upstream `SearchResponse.hits.total` type doesn't match the expected type based on https://www.elastic.co/guide/en/elasticsearch/reference/current/search-search.html

Reply:
Looks like it was a `number` before 7.0: https://www.elastic.co/guide/en/elasticsearch/reference/current/breaking-changes-7.0.html#hits-total-now-object-search-response

Reply:
The [types package](https://www.npmjs.com/package/@types/elasticsearch) is out of date by over a year.

There's an [issue](https://github.com/elastic/elasticsearch-js/issues/933) to include up-to-date types with the client, but nothing has been merged. We may be better off preferring to use our own `SearchResponse` type in general over the unmaintained `SearchResponse` from ES 5 - until there are centrally maintained response types.

Reply:
Yeah, this has been an issue for me in the past too. I think we're on our own until those are updated...